File: app/src/main/java/com/mdmobile/pocketconsole/ui/logIn/LoginConfigureSecretIdFragment.java
package com.mdmobile.pocketconsole.ui.logIn;


import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.EditText;

import com.mdmobile.pocketconsole.R;

public class LoginConfigureSecretIdFragment extends Fragment {

    public EditText clientIdEditText, apiSecretEditText;

    public LoginConfigureSecretIdFragment() {
        //Empty constructor required
    }

    public static LoginConfigureSecretIdFragment newInstance() {
        return new LoginConfigureSecretIdFragment();
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View rootView = inflater.inflate(R.layout.fragment_api_client_secret, container, false);
        clientIdEditText = rootView.findViewById(R.id.client_id_text_view);
        apiSecretEditText = rootView.findViewById(R.id.api_secret_text_view);
        return rootView;
    }
}

File: app/src/main/java/com/mdmobile/cyclops/ui/logIn/LoginConfigureSecretIdFragment.kt
package com.mdmobile.cyclops.ui.logIn


import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.EditText
import androidx.fragment.app.Fragment
import androidx.lifecycle.ViewModelProviders
import com.mdmobile.cyclops.R

class LoginConfigureSecretIdFragment : Fragment(), View.OnFocusChangeListener {

    private lateinit var clientIdEditText: EditText
    private lateinit var apiSecretEditText: EditText
    private lateinit var viewModel: LoginViewModel


    //Interfaces
    override fun onFocusChange(v: View, hasFocus: Boolean) {
        if (hasFocus) {
            return
        }
        val text = (v as EditText).text.toString()
        when (v.id) {
            R.id.instance_client_id_edit_text -> (activity as LoginActivity).viewModel.updateClientId(text)
            R.id.instance_secret_edit_text -> (activity as LoginActivity).viewModel.updateApiSecret(text)
            else -> throw IllegalArgumentException("View not known: ${v.id}")
        }

    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        viewModel = ViewModelProviders.of(this, (activity as LoginActivity).viewModelFactory).get(LoginViewModel::class.java)
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
        val rootView = inflater.inflate(R.layout.fragment_api_client_secret, container, false)
        clientIdEditText = rootView.findViewById(R.id.instance_client_id_edit_text)
        apiSecretEditText = rootView.findViewById(R.id.instance_secret_edit_text)

        clientIdEditText.onFocusChangeListener = this
        apiSecretEditText.onFocusChangeListener = this

        return rootView
    }

    companion object {

        fun newInstance(): LoginConfigureSecretIdFragment {
            return LoginConfigureSecretIdFragment()
        }
    }
}
